bases/shared/instagoric-server/server.js

```diff
@@ -668,6 +670,78 @@ faucetapp.get('/transaction-status/:txhash', (req, res) => {
   else res.status(400).send('invalid form');
 });
 
+publicapp.get('/install-bundle', async (req, res) => {
+  res.send(`
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>File Upload</title>
+</head>
+<body>
+
+    <h1>Upload a File</h1>
+
+    <form id="uploadForm" action="/install-bundle" method="POST" enctype="multipart/form-data">
+        <input type="file" name="file" id="fileInput" required>
+        <br><br>
+        <button type="submit">Upload</button>
+    </form>
+
+    <script>
+        document.getElementById('uploadForm').addEventListener('submit', function(event) {
+            event.preventDefault(); // Prevent the default form submission
+
+            const formData = new FormData();
+            const fileInput = document.getElementById('fileInput');
+            
+            if (fileInput.files.length === 0) {
+                alert("Please select a file.");
+                return;
+            }
+
+            formData.append('file', fileInput.files[0]);
+
+            fetch('/install-bundle', {
+                method: 'POST',
+                body: formData
+            })
+            .then(data => {
+                alert('File uploaded successfully!');
```

Comment:
I think we should check the response status before declaring the API request as a success. 
```js
.then(response => {
    if (!response.ok) {
        throw new Error('Failed to upload file. Server responded with status ' + response.status);
    }
    return response.json();
})
```

Reply:
It won't be a then if status is other than 200.
@frazarshad Right?